File: src/my_cpp_pkg/src/robot_news_station.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class RobotNewsStation : public rclcpp::Node {
public:
    RobotNewsStation(): rclcpp::Node("robot_news_station"), counter_(0){
        pub_ = this -> create_publisher<std_msgs::msg::String>("robot_news", 10);
        timer_ = create_wall_timer (
            std::chrono::milliseconds(500),
            std::bind(&RobotNewsStation::timerCallback, this));
    }
private:
    void timerCallback() {
        counter_++;
        auto msg = std_msgs::msg::String();
        msg.data = "New from" + std::string(get_name());
        pub_ -> publish(msg);
    }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
    int counter_;
};
int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStation>();
    rclcpp::spin(node);
    rclcpp::shutdown();

}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class RobotNewsStation(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.declare_parameter("robot_name", "my_robot")
        self.robot_name = self.get_parameter("robot_name").value

        self.pub_ = self.create_publisher(String, "robot_news", 10)
        self.timer = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot News Station started")

    def publish_news(self):
        msg = String()
        msg.data = "Breaking news from" + self.robot_name
        self.pub_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStation()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ =="__main__":
    main()

    